### PR TITLE
Inference TensorRT execution provider container revival

### PR DESCRIPTION
**Pull request by probicheaux**
# Description

Revives the trt container, bugfixing engine caching, and disables ORT graph optimizations (since trt can do them better)

Screenshot 2024-04-09 at 4 14 16 PM

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

Locally

## Any specific deployment considerations

New container push

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/models/roboflow.py [hunk 1]
```diff
@@ -604,6 +605,9 @@ def __init__(
                             "trt_fp16_enable": True,
```
Comment (PawelPeczek-Roboflow) [inline inference/core/models/roboflow.py:605]: Question: is there any value from making that parametrised by env?

File: inference/core/models/roboflow.py [hunk 2]
```diff
@@ -695,12 +699,28 @@ def initialize_model(self) -> None:
             t1_session = perf_counter()
             # Create an ONNX Runtime Session with a list of execution providers in priority order. ORT attempts to load providers until one is successful. This keeps the code across devices identical.
             providers = self.onnxruntime_execution_providers
+
             if not self.load_weights:
                 providers = ["OpenVINOExecutionProvider", "CPUExecutionProvider"]
             try:
+                has_trt = False
```
Comment (PawelPeczek-Roboflow) [inline inference/core/models/roboflow.py:706]: could we extract to separate util functiom?

File: docker/dockerfiles/Dockerfile.onnx.trt [hunk 3]
```diff
@@ -1,5 +1,4 @@
-FROM roboflow/roboflow-inference-server-trt-base:latest
-
+FROM nvcr.io/nvidia/tensorrt:22.12-py3
```
Comment (PawelPeczek-Roboflow) [inline docker/dockerfiles/Dockerfile.onnx.trt:1]: :+1

**Review by PawelPeczek-Roboflow — APPROVED** [review]

File: inference/core/models/roboflow.py (hunk 1 above)
Comment (probicheaux) [reply inference/core/models/roboflow.py:606]: Enabling this flag uniformly made inference slower even for unconverted models in my tests

**Review by tonylampada — APPROVED** [review]

**Pull request merged**